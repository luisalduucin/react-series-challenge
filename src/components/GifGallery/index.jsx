import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import {FavoriteBtn, HomeWrapper, SearchBtn, SearchInput, SearchWrapper, TrendingGifCards} from './GifGallery.style';
import searchIcon from './search.png';
import favoriteIcon from './heart-filled-white.png'
import GifCard from "../GifCard";

class GifGallery extends React.Component {
  componentWillMount() {
    if (this.props.dataSource === 'GIPHY') {
      this.props.fetchTrendingGifs();
    }
  }

  render() {
    const dataSource = this.props.dataSource === 'GIPHY' ? this.props.trending: this.props.favorites;
    return (
      <HomeWrapper>
        <SearchWrapper>
          <SearchInput placeholder={'Type to search cool GIFs'} />
          <SearchBtn src={searchIcon} />
          <FavoriteBtn src={favoriteIcon} onClick={() => this.props.history.push('/favorites')} />
        </SearchWrapper>
        <TrendingGifCards>
          {
            dataSource.map((gif) => {
              const isFavorite = !!this.props.favorites.find((favorite) => favorite.id === gif.id);
              return (
                <GifCard
                  gif={gif}
                  isFavorite={isFavorite}
                  imageSize={'small'}
                  key={gif.id}
                  toggleFavorite={this.props.toggleFavorite}
                 />
              )
            })
          }
        </TrendingGifCards>
      </HomeWrapper>
    );
  }
}

GifGallery.propTypes = {
  fetchTrendingGifs: PropTypes.func,
  toggleFavorite: PropTypes.func.isRequired,
  trending: PropTypes.array.isRequired
};

export default withRouter(GifGallery);
