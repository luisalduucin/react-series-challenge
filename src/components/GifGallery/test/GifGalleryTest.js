import React from 'react';
import GifGallery from "../index";
import {shallow} from "enzyme";

const emptyDataSource = [];
const mockGif = {
  id: 'p3n7NWvfz3u0c1meay',
  images: {
    original: 'https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif',
    small: 'https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif'
  }
};
const mockDataSource = [mockGif];

describe('GifCard component', () => {
  it('should render', () => {
    const component = shallow(
      <GifGallery.WrappedComponent
        dataSource={'favorites'}
        favorites={mockDataSource}
        fetchTrendingGifs={jest.fn()}
        trending={emptyDataSource}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render', () => {
    const component = shallow(
      <GifGallery.WrappedComponent
        dataSource={'GIPHY'}
        favorites={emptyDataSource}
        fetchTrendingGifs={jest.fn()}
        trending={mockDataSource}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
