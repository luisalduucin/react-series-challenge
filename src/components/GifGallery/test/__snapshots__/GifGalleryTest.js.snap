// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifCard component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GifGallery
    dataSource="favorites"
    favorites={
        Array [
            Object {
              "id": "p3n7NWvfz3u0c1meay",
              "images": Object {
                "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
              },
            },
          ]
    }
    fetchTrendingGifs={[Function]}
    trending={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.img
                    src={
                              Object {
                                        "process": [Function],
                                      }
                    }
          />
          <styled.input
                    placeholder="Type to search cool GIFs"
          />
          <styled.img
                    onClick={[Function]}
                    src={
                              Object {
                                        "process": [Function],
                                      }
                    }
          />
          <styled.img
                    onClick={[Function]}
                    src={
                              Object {
                                        "process": [Function],
                                      }
                    }
          />
</styled.div>,
        <styled.div>
          <Connect(withRouter(GifCard))
                    gif={
                              Object {
                                        "id": "p3n7NWvfz3u0c1meay",
                                        "images": Object {
                                          "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                                          "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                                        },
                                      }
                    }
                    imageSize="small"
                    isDetailBtnVisible={true}
                    isFavorite={true}
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.img
              src={
                            Object {
                                          "process": [Function],
                                        }
              }
/>,
            <styled.input
              placeholder="Type to search cool GIFs"
/>,
            <styled.img
              onClick={[Function]}
              src={
                            Object {
                                          "process": [Function],
                                        }
              }
/>,
            <styled.img
              onClick={[Function]}
              src={
                            Object {
                                          "process": [Function],
                                        }
              }
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": Object {
                "process": [Function],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "placeholder": "Type to search cool GIFs",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClick": [Function],
              "src": Object {
                "process": [Function],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClick": [Function],
              "src": Object {
                "process": [Function],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Connect(withRouter(GifCard))
              gif={
                            Object {
                                          "id": "p3n7NWvfz3u0c1meay",
                                          "images": Object {
                                            "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                                            "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                                          },
                                        }
              }
              imageSize="small"
              isDetailBtnVisible={true}
              isFavorite={true}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "p3n7NWvfz3u0c1meay",
            "nodeType": "class",
            "props": Object {
              "gif": Object {
                "id": "p3n7NWvfz3u0c1meay",
                "images": Object {
                  "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                  "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                },
              },
              "imageSize": "small",
              "isDetailBtnVisible": true,
              "isFavorite": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.img
                        src={
                                    Object {
                                                "process": [Function],
                                              }
                        }
            />
            <styled.input
                        placeholder="Type to search cool GIFs"
            />
            <styled.img
                        onClick={[Function]}
                        src={
                                    Object {
                                                "process": [Function],
                                              }
                        }
            />
            <styled.img
                        onClick={[Function]}
                        src={
                                    Object {
                                                "process": [Function],
                                              }
                        }
            />
</styled.div>,
          <styled.div>
            <Connect(withRouter(GifCard))
                        gif={
                                    Object {
                                                "id": "p3n7NWvfz3u0c1meay",
                                                "images": Object {
                                                  "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                                                  "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                                                },
                                              }
                        }
                        imageSize="small"
                        isDetailBtnVisible={true}
                        isFavorite={true}
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.img
                src={
                                Object {
                                                "process": [Function],
                                              }
                }
/>,
              <styled.input
                placeholder="Type to search cool GIFs"
/>,
              <styled.img
                onClick={[Function]}
                src={
                                Object {
                                                "process": [Function],
                                              }
                }
/>,
              <styled.img
                onClick={[Function]}
                src={
                                Object {
                                                "process": [Function],
                                              }
                }
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "src": Object {
                  "process": [Function],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "placeholder": "Type to search cool GIFs",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClick": [Function],
                "src": Object {
                  "process": [Function],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClick": [Function],
                "src": Object {
                  "process": [Function],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Connect(withRouter(GifCard))
                gif={
                                Object {
                                                "id": "p3n7NWvfz3u0c1meay",
                                                "images": Object {
                                                  "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                                                  "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                                                },
                                              }
                }
                imageSize="small"
                isDetailBtnVisible={true}
                isFavorite={true}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "p3n7NWvfz3u0c1meay",
              "nodeType": "class",
              "props": Object {
                "gif": Object {
                  "id": "p3n7NWvfz3u0c1meay",
                  "images": Object {
                    "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                    "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                  },
                },
                "imageSize": "small",
                "isDetailBtnVisible": true,
                "isFavorite": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`GifCard component should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GifGallery
    dataSource="GIPHY"
    favorites={Array []}
    fetchTrendingGifs={[Function]}
    trending={
        Array [
            Object {
              "id": "p3n7NWvfz3u0c1meay",
              "images": Object {
                "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.img
                    src={
                              Object {
                                        "process": [Function],
                                      }
                    }
          />
          <styled.input
                    placeholder="Type to search cool GIFs"
          />
          <styled.img
                    onClick={[Function]}
                    src={
                              Object {
                                        "process": [Function],
                                      }
                    }
          />
          <styled.img
                    onClick={[Function]}
                    src={
                              Object {
                                        "process": [Function],
                                      }
                    }
          />
</styled.div>,
        <styled.div>
          <Connect(withRouter(GifCard))
                    gif={
                              Object {
                                        "id": "p3n7NWvfz3u0c1meay",
                                        "images": Object {
                                          "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                                          "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                                        },
                                      }
                    }
                    imageSize="small"
                    isDetailBtnVisible={true}
                    isFavorite={false}
          />
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.img
              src={
                            Object {
                                          "process": [Function],
                                        }
              }
/>,
            <styled.input
              placeholder="Type to search cool GIFs"
/>,
            <styled.img
              onClick={[Function]}
              src={
                            Object {
                                          "process": [Function],
                                        }
              }
/>,
            <styled.img
              onClick={[Function]}
              src={
                            Object {
                                          "process": [Function],
                                        }
              }
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": Object {
                "process": [Function],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "placeholder": "Type to search cool GIFs",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClick": [Function],
              "src": Object {
                "process": [Function],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClick": [Function],
              "src": Object {
                "process": [Function],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Connect(withRouter(GifCard))
              gif={
                            Object {
                                          "id": "p3n7NWvfz3u0c1meay",
                                          "images": Object {
                                            "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                                            "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                                          },
                                        }
              }
              imageSize="small"
              isDetailBtnVisible={true}
              isFavorite={false}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "p3n7NWvfz3u0c1meay",
            "nodeType": "class",
            "props": Object {
              "gif": Object {
                "id": "p3n7NWvfz3u0c1meay",
                "images": Object {
                  "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                  "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                },
              },
              "imageSize": "small",
              "isDetailBtnVisible": true,
              "isFavorite": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.img
                        src={
                                    Object {
                                                "process": [Function],
                                              }
                        }
            />
            <styled.input
                        placeholder="Type to search cool GIFs"
            />
            <styled.img
                        onClick={[Function]}
                        src={
                                    Object {
                                                "process": [Function],
                                              }
                        }
            />
            <styled.img
                        onClick={[Function]}
                        src={
                                    Object {
                                                "process": [Function],
                                              }
                        }
            />
</styled.div>,
          <styled.div>
            <Connect(withRouter(GifCard))
                        gif={
                                    Object {
                                                "id": "p3n7NWvfz3u0c1meay",
                                                "images": Object {
                                                  "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                                                  "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                                                },
                                              }
                        }
                        imageSize="small"
                        isDetailBtnVisible={true}
                        isFavorite={false}
            />
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.img
                src={
                                Object {
                                                "process": [Function],
                                              }
                }
/>,
              <styled.input
                placeholder="Type to search cool GIFs"
/>,
              <styled.img
                onClick={[Function]}
                src={
                                Object {
                                                "process": [Function],
                                              }
                }
/>,
              <styled.img
                onClick={[Function]}
                src={
                                Object {
                                                "process": [Function],
                                              }
                }
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "src": Object {
                  "process": [Function],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "placeholder": "Type to search cool GIFs",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClick": [Function],
                "src": Object {
                  "process": [Function],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onClick": [Function],
                "src": Object {
                  "process": [Function],
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Connect(withRouter(GifCard))
                gif={
                                Object {
                                                "id": "p3n7NWvfz3u0c1meay",
                                                "images": Object {
                                                  "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                                                  "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                                                },
                                              }
                }
                imageSize="small"
                isDetailBtnVisible={true}
                isFavorite={false}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "p3n7NWvfz3u0c1meay",
              "nodeType": "class",
              "props": Object {
                "gif": Object {
                  "id": "p3n7NWvfz3u0c1meay",
                  "images": Object {
                    "original": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/giphy.gif",
                    "small": "https://media3.giphy.com/media/p3n7NWvfz3u0c1meay/200w_d.gif",
                  },
                },
                "imageSize": "small",
                "isDetailBtnVisible": true,
                "isFavorite": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
